ui/app/services/wizard.js

```diff
@@ -268,18 +318,24 @@ export default Service.extend(DEFAULTS, {
     } else {
       this.saveExtState(
         STORAGE_KEYS.COMPLETED_FEATURES,
-        this.getExtState(STORAGE_KEYS.COMPLETED_FEATURES).toArray().addObject(done)
+        this.getExtState(STORAGE_KEYS.COMPLETED_FEATURES)
+          .toArray()
+          .addObject(done)
       );
     }
 
     this.saveExtState(STORAGE_KEYS.FEATURE_LIST, features.length ? features : null);
     this.storage().removeItem(STORAGE_KEYS.FEATURE_STATE);
+    if (this.featureMachineHistory) {
+      this.set('featureMachineHistory', []);
```

Comment:
Is the default empty array or `null`? Or is there a reason for it not to be null here?

Reply:
yes. this is how we differentiate between "there is no feature history right now" and "there has never been feature history before". If we leave it ambiguous then progressing onto the next feature will get rid of the progress bar altogether.